shared/src/commonMain/kotlin/org/hyperskill/app/user_storage/cache/UserStorageCacheDataSourceImpl.kt

```diff
@@ -0,0 +1,27 @@
+package org.hyperskill.app.user_storage.cache
+
+import com.russhwolf.settings.Settings
+import com.russhwolf.settings.get
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import org.hyperskill.app.user_storage.data.source.UserStorageCacheDataSource
+import org.hyperskill.app.user_storage.domain.model.UserStorage
+
+class UserStorageCacheDataSourceImpl(
+    private val json: Json,
+    private val settings: Settings
+) : UserStorageCacheDataSource {
+    override fun getUserStorage(): Result<UserStorage> =
+        kotlin.runCatching {
+            json.decodeFromString(settings[UserStorageCacheKeyValues.USER_STORAGE, ""])
```

Comment:
Давай как и во всем проекте использовать `get` методы:

```suggestion
            json.decodeFromString(settings.getString(UserStorageCacheKeyValues.USER_STORAGE, ""))
```